Auto translate

quinnha opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Improved `translateComments` with various fixes

## Description
Optimized `translateComments` to better target comment components and eliminate language detect API calls since translating costs the same (and can detect languages!)

Added one translation tag per comment at the bottom of the comment (see issue #4)

Also cleaned up code a little bit for readability (too many brackets lol), and simplified where needed.

![](https://github.com/wastadtlander/Youtube-Comment-Translator/assets/35006641/7466d6d8-2298-418b-b2d1-1584d8d93db4)

quinnha committed:
removed language detection

quinnha committed:
targeted child nodes for translation to avoid nuking links

quinnha committed:
added translation message at the bottom of comments

quinnha committed:
minor formatting edits

quinnha committed:
minor formatting edits

quinnha committed:
minor comments

quinnha committed:
removed my key LOL

wastadtlander commented on scripts/content.js:
```diff
@@ -45,56 +62,63 @@ function callTranslateAPI(text) {
     if (mutation.type == "childList") {
-      mutation.addedNodes.forEach((node) => {
+      for (const node of mutation.addedNodes) {
         if (node.nodeName.toLowerCase() == "ytd-comment-renderer") {
```
Why is the `toLowerCase` method required here?

quinnha replied:
good question, will remove

quinnha replied:
`toLowerCase` helps with readibility, the classes are written in lowercase, so just helps with matching

alternatively, we can just do: `node.nodeName == "YTD-COMMENT-RENDERER"` but that is ugly, let me know which one you want

wastadtlander replied:
yeah u right caps is ugly.

wastadtlander commented on scripts/content.js:
```diff
@@ -106,20 +130,24 @@ commentWrapper.addEventListener("click", (event) => {
 
 /**
- * Observer to translate new comments on load
+ * Observer setup
```
One line this.

quinnha replied:
boooo

wastadtlander reviewed (approved):
Looks nice.

quinnha committed:
fixed small issue with translation tag, address comments

quinnha committed:
changed == to === and != to ===

quinnha committed:
removed my key because im dumb (if ur looking in the future, this key doesnt work anymore)

quinnha committed:
updated comment

quinnha commented:
good to merge, just fixed a small issue with translation tag showing when no translation occurred

also removed detectURL since we arent using it anymore
